### PR TITLE
DSA Assignment on list first submission

### PR DESCRIPTION
**Pull request by meta-manoj-sharma**

### CODE REVIEW

File: DSA_Session_1/SubListRotation/src/sublist/LinkedList.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+	 * @param start
+	 * @param end
+	 * @param rotateIndex
+	 */
+	private void validate(int startOfSubList, int endOfSubList, int noOfRotation) {
+
+		if (startOfSubList <= 0) {
+			throw new AssertionError("Start Index Must Be Positive Number");
+		}
+		if (endOfSubList >= size()) {
```
Comment (jbhavya) [inline DSA_Session_1/SubListRotation/src/sublist/LinkedList.java]: Why the end of sublist can't be the last element of the list? Modify the logic such as sublist should take the last element also

File: DSA_Session_1/SubListRotation/src/sublist/LinkedList.java [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+		}
+	}
+
+	/**
+	 * 
+	 * @param start
+	 * @param end
+	 * @param rotateIndex
+	 */
+	private void validate(int startOfSubList, int endOfSubList, int noOfRotation) {
```
Comment (jbhavya) [inline DSA_Session_1/SubListRotation/src/sublist/LinkedList.java]: What is the use of variable 'noOfRotation'? Remove unused variables.

File: DSA_Session_1/SubListRotation/src/sublist/LinkedList.java [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+	}
+
+	/**
+	 * @return size of Linked List
+	 */
+	public int size() {
+		int size = 0;
+		Node current = headNode;
+		while (current != null) {
+
```
Comment (jbhavya) [inline DSA_Session_1/SubListRotation/src/sublist/LinkedList.java]: Remove extra lines where not needed.

File: DSA_Session_1/SubListRotation/src/sublist/LinkedList.java [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+	/**
+	 * This method checks whether there is a loop in a linked list or not
+	 * @param list is the linked list in which loop is to be found
+	 * @return true when there exists a loop in a linked list otherwise returns false
+	 */
+	public boolean detectLoopInList(Node headNode, LinkedList list) {
+		Set<Node> referenceSet = new HashSet<Node>();
+		Node traverse = list.headNode;
+		while (traverse != null) {
+			if (referenceSet.contains(traverse)) {// Checking whether the node's
```
Comment (jbhavya) [inline DSA_Session_1/SubListRotation/src/sublist/LinkedList.java]: Commentation should be proper.

File: DSA_Session_1/SubListRotation/src/sublist/Main.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+	public static void main(String[] args) {
+		try {
+
+			Scanner input = new Scanner(System.in);
+			int size, choice, row, start, end, noOfRotation;
+			while (true) {
+				LinkedList list = new LinkedList();
+				System.out.println("Enter size of linked list");
+				size = input.nextInt();
+				if (size == 0) {
```
Comment (jbhavya) [inline DSA_Session_1/SubListRotation/src/sublist/Main.java]: What if the size is less than 0?

File: DSA_Session_1/SubListRotation/src/sublist/LinkedList.java (hunk 1 above)
Comment (meta-manoj-sharma) [reply DSA_Session_1/SubListRotation/src/sublist/LinkedList.java]: issue is resolved.

File: DSA_Session_1/SubListRotation/src/sublist/LinkedList.java (hunk 2 above)
Comment (meta-manoj-sharma) [reply DSA_Session_1/SubListRotation/src/sublist/LinkedList.java]: extra parameter is removed.

File: DSA_Session_1/SubListRotation/src/sublist/Main.java (hunk 5 above)
Comment (meta-manoj-sharma) [reply DSA_Session_1/SubListRotation/src/sublist/Main.java]: issue resolved

File: DSA_Session_1/SubListRotation/src/sublist/LinkedList.java (hunk 3 above)
Comment (meta-manoj-sharma) [reply DSA_Session_1/SubListRotation/src/sublist/LinkedList.java]: resolved

File: DSA_Session_1/SubListRotation/src/sublist/LinkedList.java (hunk 4 above)
Comment (meta-manoj-sharma) [reply DSA_Session_1/SubListRotation/src/sublist/LinkedList.java]: resolved

**Pull request merged**